internal/crosstest/cross_test.go

```diff
@@ -69,7 +70,10 @@ func (c crossServerReRPC) Fail(ctx context.Context, req *crosspb.FailRequest) (*
 	return nil, rerpc.Errorf(rerpc.CodeResourceExhausted, errMsg)
 }
 
-func (c crossServerReRPC) Sum(ctx context.Context, stream *crosspb.CrossServiceReRPC_Sum) error {
+func (c crossServerReRPC) Sum(
+	ctx context.Context,
+	stream *handlerstream.Client[crosspb.SumRequest, crosspb.SumResponse],
+) error {
```

Comment:
This is a good example of how user's code changes. On the client side, the stream type is nearly always inferred (it's usually returned from a client method, so the user doesn't type it out). Users *do* need to write the stream's type in their service implementations, though.

To me, this is a little wordier than the generated types, but also clearer - it's obvious that the different client streams are all the same type, and that they'll have the same method set. The names are also (maybe?) better than the weirdo generated names with underscores: `*crosspb.CrossServiceReRPC_Sum` vs `*handlerstream.Client[crosspb.SumRequest, crosspb.SumResponse]`.